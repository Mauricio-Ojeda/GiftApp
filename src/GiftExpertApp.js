import React, { useState } from 'react'
import AddCategory from './components/AddCategory'
import GifGrid from './components/GifGrid'

const GiftExpertApp = ({ defaultCategories = [] }) => {

    const [categories, setCategories] = useState( defaultCategories )

    console.log(categories);

    
    return (
        <>
            <h2 className="animate__fadeInDownBig animate__fadeOutDownBig">GiftExpertApp</h2>
            <AddCategory 
                setCategories = { setCategories }
            />
            <hr/>            
            <ol>
                {
                    categories.map( (category, index ) => {
                      return  <GifGrid 
                            key = { category + index }
                            category = { category }
                        />
                    })
                }
                
            </ol>      

        </>
    )
}

export default GiftExpertApp
