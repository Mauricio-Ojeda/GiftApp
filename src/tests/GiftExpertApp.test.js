import { shallow } from "enzyme";
import GiftExpertApp from "../GiftExpertApp";

describe('pruebas en <GiftExpertApp />', () => {

    
    test('Crear snapshot para detectar cambios en el codigo', () => {
        
        const wrapper = shallow(<GiftExpertApp />);
        expect(wrapper).toMatchSnapshot();
        
    });
    
    test('debe de mostrar una lista de categorias', () => {
        const categories = [ 'Messi', 'Programador' ];
        const wrapper = shallow(<GiftExpertApp defaultCategories= { categories } />);

        expect(wrapper).toMatchSnapshot();
        expect( wrapper.find('GifGrid').length).toBe(categories.length);
        

    })
    

})
