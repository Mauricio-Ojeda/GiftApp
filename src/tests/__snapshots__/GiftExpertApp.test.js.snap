// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en <GiftExpertApp /> Crear snapshot para detectar cambios en el codigo 1`] = `
<Fragment>
  <h2
    className="animate__fadeInDownBig animate__fadeOutDownBig"
  >
    GiftExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`pruebas en <GiftExpertApp /> debe de mostrar una lista de categorias 1`] = `
<Fragment>
  <h2
    className="animate__fadeInDownBig animate__fadeOutDownBig"
  >
    GiftExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Messi"
      key="Messi0"
    />
    <GifGrid
      category="Programador"
      key="Programador1"
    />
  </ol>
</Fragment>
`;
